### DISCUSSION TITLE
Possible to show Wearout of SSD as sensors?

### DISCUSSION THREAD

**Discussion by DieMetRik** [dougiteixeira/proxmoxve#302] [Feature request] [answered]
### Checklist

- [X] I have updated to the latest available Proxmox VE Custom Integration version, [see the latest version here](https://github.com/dougiteixeira/proxmoxve/releases/latest).

### Describe the issue you are experiencing

Very useful integration thanks a lot. Possible to ADD next:

1. Is possible to show Wearout of SSD as sensors?
![image](https://github.com/dougiteixeira/proxmoxve/assets/31734768/076cb6c9-f2db-458b-83df-8426b264c61a)

### In which version of Home Assistant Core do you have the problem?

2024.04.02

### What version of Proxmox VE Custom Integration has the issue?

3.4.1

### What version of Proxmox VE do you have the problem?

8.2.2

**Comment by dougiteixeira** [comment]
@DieMetRik 
Please attach the diagnostics file so that I can evaluate the available information about this disk attribute.

I'm waiting for the shipment to oblige with this.

**Reply by DieMetRik** [reply]
How I can get this file?

**Comment by dougiteixeira** [comment]
Please, you can follow the steps below:
* [Enable integration debugging](https://www.home-assistant.io/docs/configuration/troubleshooting/#enabling-debug-logging).
* Reload the integration configuration entry.
* [Download a diagnostic file](https://www.home-assistant.io/docs/configuration/troubleshooting/#download-diagnostics).
* [Disable integration debugging and download the generated log file](https://www.home-assistant.io/docs/configuration/troubleshooting/#disable-debug-logging-and-download-logs).
* Attach the diagnostic file and the generated log file here.

**Reply by DieMetRik** [reply]
Here
[home-assistant_proxmoxve_2024-05-19T18-31-16.850Z.log](https://github.com/dougiteixeira/proxmoxve/files/15370414/home-assistant_proxmoxve_2024-05-19T18-31-16.850Z.log)

**Reply by dougiteixeira** [reply]
The [diagnostic file](https://www.home-assistant.io/docs/configuration/troubleshooting/#download-diagnostics) was missing... you only attached the log file.

**Comment by dougiteixeira** [comment, marked as answer]
Added in #305. It will be available in the next version (3.4.2).

**Comment by DieMetRik** [comment]
Thanks a lot for your job!
All wearout show 0%
![image](https://github.com/dougiteixeira/proxmoxve/assets/31734768/1df214f4-6b60-41e5-83f9-67986a73f3fa)

from my file:

vme0n1', 'wwn': 'eui.5cd2e48711b1445d', 'used': 'BIOS boot', 'serial': 'BTKA129612HT512A', 'by_id_link': '/dev/disk/by-id/nvme-670p_SSDPEKNU512GZ_NVMe_INTEL_512GB_BTKA129612HT512A', 'osdid-list': None, 'model': '670p SSDPEKNU512GZ NVMe INTEL 512GB', 'osdid': -1, 'health': 'PASSED', 'rpm': 0, 'size': 512110190592, 'vendor': 'unknown', '`wearout': 90`}, {'type': 'nvme', 'devpath': '/dev/nvme1n1', 'gpt': 0, 'by_id_link': '/dev/disk/by-id/nvme-Samsung_SSD_980_500GB_S78GNL0X224771D_1', 'wwn': 'eui.002538d241a0d533', 'used': 'LVM', 'serial': 'S78GNL0X224771D', 'model': 'Samsung SSD 980 500GB', 'osdid-list': None, 'osdid': -1, 'rpm': 0, 'size': 500107862016, 'health': 'PASSED', '`wearout': 100`, 'vendor': 'unknown'}, {'wwn': '0x5000c50090be786d', 'used': 'ext4', 'serial': 'Z840G2BF', 'by_id_link': '/dev/disk/by-id/ata-ST8000AS0002-1NA17Z_Z840G2BF', 'osdid-list': None, 'model': 'ST8000AS0002-1NA17Z', 'gpt': 1, 'type': 'hdd', 'devpath': '/dev/sda', 'vendor': 'ATA     ', 'wearout': 'N/A', 'osdid': -1, 'health': 'PASSED', 'rpm': '5980', 'size': 8001563222016}, {'osdid': -1, 'health': 'PASSED', 'rpm': 0, 'size': 250059350016, 'vendor': 'ATA     ', '`wearout': 96`, 'gpt': 0, 'devpath': '/dev/sdb', 'type': 'ssd', 'serial': 'S3Y9NX0K750133E', 'wwn': '0x5002538e4060dbee', 'used': 'LVM', 'by_id_link': '/dev/disk/by-id/ata-Samsung_SSD_860_EVO_250GB_S3Y9NX0K750133E', 'osdid-list': None, 'model': 'Samsung_SSD_860_EVO_250GB'}, {'vendor': 'ATA     ', '`wearout': 97`, 'health': 'PASSED', 'size': 256060514304, 'rpm': 0, 'osdid': -1, 'osdid-list': None, 'model': 'RTNRB256RFM4KWDL', 'serial': 'SD0M61139R1SZ6CL12HS', 'used': 'ext4', 'wwn': '0x5044a5006c281200', 'by_id_link': '/dev/disk/by-id/ata-RTNRB256RFM4KWDL_SD0M61139R1SZ6CL12HS', 'gpt': 1, 'devpath': '/dev/sdc', 'type': 'ssd'}, {'wwn': '0x5000c500e4f2fbb6', 'used': 'ext4', 'serial': 'ZR5AR6WD', 'by_id_link': '/dev/disk/by-id/ata-ST18000NM000J-2TV103_ZR5AR6WD', 'osdid-list': None, 'model': 'ST18000NM000J-2TV103', 'gpt': 1, 'type': 'hdd', 'devpath': '/dev/sdd', 'vendor': 'ATA     ', 'wearout': 'N/A', 'osdid': -1, 'health': 'PASSED', 'size': 18000207937536, 'rpm': '7200'}]

**Comment by DieMetRik** [comment]
Need to create a new issue?

**Comment by Lipown** [comment]
I guess to get the wear out i need to enable this:
![image](https://github.com/user-attachments/assets/172a27af-d8d3-4346-8310-06ea8b028a1c)

That will keep my rotational disks spinning up unnecessarily.
1) Can I get wear out some different way please?
2) I would like to get also physical disks info but without spinning that up - would it be able to do the check whether the drive spinning and only if so to pull the data? 

Thanks!
